annotated.go

```diff
@@ -312,6 +316,33 @@ func (ann *annotated) Build() (interface{}, error) {
 	return newFn.Interface(), nil
 }
 
+// checks whether the target function is either
+// returning an fx.Out struct or an taking in a
+// fx.In struct as a parameter.
+func (ann *annotated) typeCheckOrigFn() error {
+	ft := reflect.TypeOf(ann.Target)
+	for i := 0; i < ft.NumOut(); i++ {
+		ot := ft.Out(i)
+		if ot.Kind() != reflect.Struct {
+			continue
+		}
+		if ot.NumField() > 0 && ot.Field(0).Type == reflect.TypeOf(Out{}) {
+			return fmt.Errorf("fx.Out structs cannot be annotated")
```

Comment:
Consider defining a const for this error. It could be reused in the test during assertion.

Reply:
```suggestion
			return errors.New("fx.Out structs cannot be annotated")
```